### ISSUE TITLE
TIP import: support iterated matches on the same variable

### ISSUE THREAD

**Issue by jannikvierling** [gapt/gapt#585]
Running `TipSmtParser.fixupAndParse` on the file `./../benchmarks/benchmarks/tip2015/regexp_RecPlus.smt2` results in a `java.lang.IllegalArgumentException`.

Example:
```
import at.logic.gapt.formats.tip.TipSmtParser
import better.files._

object Bug {
  def main(args: Array[String]): Unit = {
    TipSmtParser fixupAndParse "./../benchmarks/benchmarks/tip2015/regexp_RecPlus.smt2".toFile
  }
}
```
Output:
```
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:212)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:97)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:94)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:90)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:90)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.parseFunctionBody(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:103)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:94)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:90)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:90)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.parseFunctionBody(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:103)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:94)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1$1.apply(tipsmt.scala:90)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.at$logic$gapt$formats$tip$TipSmtParser$$handleCase$1(tipsmt.scala:90)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parseFunctionBody$1.apply(tipsmt.scala:109)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at at.logic.gapt.formats.tip.TipSmtParser.parseFunctionBody(tipsmt.scala:109)
	at at.logic.gapt.formats.tip.TipSmtParser.parse(tipsmt.scala:61)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parse$4.apply(tipsmt.scala:156)
	at at.logic.gapt.formats.tip.TipSmtParser$$anonfun$parse$4.apply(tipsmt.scala:156)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at at.logic.gapt.formats.tip.TipSmtParser$.parse(tipsmt.scala:156)
	at at.logic.gapt.formats.tip.TipSmtParser$.fixupAndParse(tipsmt.scala:161)
```
A similar behaviour occurs when running `TipSmtParser.fixupAndParse` on the files below:
```
benchmarks/tip2015/regexp_RecPlus.smt2
benchmarks/tip2015/regexp_RecEps.smt2
benchmarks/tip2015/regexp_RecStar.smt2
benchmarks/tip2015/regexp_SeqDistrPlus.smt2
benchmarks/tip2015/regexp_RecSeq.smt2
benchmarks/tip2015/regexp_Reverse.smt2
benchmarks/tip2015/regexp_RecNil.smt2
benchmarks/tip2015/regexp_RecAtom.smt2
benchmarks/tip2015/regexp_Deeps.smt2
benchmarks/tip2015/regexp_Star.smt2
benchmarks/tip2015/regexp_PlusIdempotent.smt2
benchmarks/tip2015/regexp_PlusAssociative.smt2
benchmarks/tip2015/regexp_PlusCommutative.smt2
benchmarks/tip2015/regexp_SeqAssociative.smt2
```

**Comment by gebner** [comment]
This is the match that causes the problem:
```smt
(define-fun
  seq
    ((x R) (y R)) R
    (match x
      (case default
        (match y
          (case default
            (match x
              (case default
                (match y
                  (case default (Seq x y))
                  (case Eps x)))
              (case Eps y)))
          (case Nil Nil)))
      (case Nil Nil)))
```

**Comment by jannikvierling** [comment]
Fixed by PR #705.
